src/components/Swap/Price.tsx

```diff
@@ -54,7 +56,14 @@ export default function Price({ trade, outputUSDC }: PriceProps) {
   const [exchangeRate, usdcPrice] = useTradeExchangeRate(trade, outputUSDC, defaultBase ? 'input' : 'output')
 
   return (
-    <TextButton color="primary" onClick={onClick}>
+    <TextButton
+      color="primary"
+      onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+        onClick()
+        const el = e.target as HTMLDivElement
```

Comment:
You should not be passing state using CSS classes. Use controlled components instead.

Reply:
updated the approach here to not use CSS classes. 

i'm not sure i understand your Controlled Components suggestion. the goal here is not to pass state, but just to prevent nested onClicks from firing without completely preventing the click event from propagating to the rest of the page (i.e. we don't want to call `stopPropagation` like we do in `interface` - see Tina's comment above.

i think the new approach is better than using CSS classes! let me know if you have a specific implementation in mind here.